litmus-portal/cluster-agents/event-tracker/pkg/utils/utils.go

```diff
@@ -0,0 +1,246 @@
+		return key < value
+	} else if operator == "GreaterThan" {
+		return key > value
+	} else if operator == "GreaterThanEqualTo" {
+		return key >= value
+	} else if operator == "LessThanEqualTo" {
+		return key <= value
+	}
+	return false
+}
+
+func conditionChecker(etp litmuschaosv1.EventTrackerPolicy, data interface{}) bool {
+	final_result := false
+	if etp.Spec.ConditionType == "and" {
+		for _, condition := range etp.Spec.Conditions {
+			result, err := jmespath.Search(condition.Key, data)
+			if err != nil {
+				log.Print(err)
+				return false
+			}
+
+			str := fmt.Sprintf("%v", result)
+			if val := cases(str, condition.Value, condition.Operator); !val {
+				final_result = val
+				break
+			} else if val {
+				final_result = true
+			}
+		}
+	} else if etp.Spec.ConditionType == "or" {
+		for _, condition := range etp.Spec.Conditions {
+			result, err := jmespath.Search(condition.Key, data)
+			if err != nil {
+				log.Print(err)
+			}
+
+			str := fmt.Sprintf("%v", result)
+			if val := cases(str, condition.Value, condition.Operator); val {
+				final_result = val
+			}
+		}
+	}
+
+	return final_result
+}
+
+func PolicyAuditor(resourceType string, obj interface{}, workflowid string) error {
+	restConfig, err := k8s.GetKubeConfig()
+	if err != nil {
+		return err
+	}
+
+	clientSet, err := dynamic.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	deploymentRes := schema.GroupVersionResource{Group: "eventtracker.litmuschaos.io", Version: "v1", Resource: "eventtrackerpolicies"}
+	deploymentConfigList, err := clientSet.Resource(deploymentRes).Namespace(AgentNamespace).List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, ep := range deploymentConfigList.Items {
+
+		eventTrackerPolicy, err := clientSet.Resource(deploymentRes).Namespace(AgentNamespace).Get(ep.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		var etp litmuschaosv1.EventTrackerPolicy
+		data, err := json.Marshal(eventTrackerPolicy.Object)
+		if err != nil {
+			return err
+		}
+
+		err = json.Unmarshal(data, &etp)
+		if err != nil {
+			return err
+		}
+
+		var dataInterface interface{}
+		var resourceName string
+		if strings.ToLower(resourceType) == "deployment" {
+			deps := obj.(*v1.Deployment)
+			resourceName = deps.GetName()
+			mar, err := json.Marshal(deps)
+			if err != nil {
+				return err
+			}
+
+			err = json.Unmarshal(mar, &dataInterface)
+		} else if strings.ToLower(resourceType) == "statefulset" {
+			sts := obj.(*v1.StatefulSet)
+			resourceName = sts.GetName()
+			mar, err := json.Marshal(sts)
+			if err != nil {
+				return err
+			}
+
+			err = json.Unmarshal(mar, &dataInterface)
+		} else if strings.ToLower(resourceType) == "daemonset" {
+			sts := obj.(*v1.DaemonSet)
+			resourceName = sts.GetName()
+			mar, err := json.Marshal(sts)
+			if err != nil {
+				return err
+			}
+
+			err = json.Unmarshal(mar, &dataInterface)
+		} else {
+			return errors.New("Resource not supported")
+		}
+
```

Comment:
you may use switch case here so the string operation is performed only once

Reply:
Done